resdk/analysis/prepare_geo.py

```diff
@@ -0,0 +1,160 @@
+"""Prepare GEO."""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import time
+import traceback
+
+from resdk.resources.utils import (
+    get_resource_collection, get_samples, get_resolwe
+)
+
+__all__ = ('prepare_geo_chipseq', 'prepare_geo_rnaseq', 'prepare_geo')
+
+
+def is_background(sample):
+    """Return ``True`` if given sample is background and ``False`` otherwise."""
+    background_relations = sample.resolwe.relation.filter(
+        type='compare',
+        label='background',
+        entity=sample.id,
+        position='background'
+    )
+    return len(background_relations) > 0
+
+
+def get_name_collection(collection_ids, res):
+    """Get the ``name`` input for ``prepare_geo`` processes."""
+    collection = None
+    if len(collection_ids) == 1:
```

Comment:
`len(collection_ids) == 1 and collection_ids[0]`

Reply:
`collection_ids` is a set and does not support indexing

Reply:
Then convert it to a list.